### PR TITLE
Remove Guava dependencies

### PR DESCRIPTION
**Pull request by jonatan-ivanov**
Related: https://github.com/wavefrontHQ/wavefront-spring-boot/issues/123

### CODE REVIEW

File: wavefront-spring-boot/src/main/java/com/wavefront/spring/autoconfigure/WavefrontSleuthSpanHandler.java [hunk 1]
```diff
@@ -403,4 +404,30 @@ private static byte decodeByte(char hi, char lo) {
     }
     return result;
   }
+
+  // Mostly from Guava
+  private static <T> T getAt(Iterator<T> iterator, int position) {
```
Comment (snicoll) [inline wavefront-spring-boot/src/main/java/com/wavefront/spring/autoconfigure/WavefrontSleuthSpanHandler.java:409]: Couldn't we get rid of this with a simple pojo structure with the indexed information that we want in a list or something.

**Review by snicoll — COMMENTED** [review]
I'd rather not copy the guava code and adapt the algorithm. Iterating on the `entrySet` every time you handle a `SpanLog` feels odd to me.

File: wavefront-spring-boot/src/main/java/com/wavefront/spring/autoconfigure/WavefrontSleuthSpanHandler.java (hunk 1 above)
Comment (jonatan-ivanov) [reply wavefront-spring-boot/src/main/java/com/wavefront/spring/autoconfigure/WavefrontSleuthSpanHandler.java]: I'm not sure we need that, please check my last commit.
On the other hand: now that I looked into what is really going on around these calls, I'm not sure I understand what was happening.

@marcingrzejszczak Could you please check if this makes sense to you too?

**Review by marcingrzejszczak — APPROVED** [review]
I think it looks OK

**Review by oppegard — APPROVED** [review]
lgtm

**Pull request merged**